base_layer/core/src/consensus/consensus_manager.rs

```diff
@@ -32,16 +40,29 @@ use crate::{
     proof_of_work::PowAlgorithm,
     proof_of_work::TargetDifficultyWindow,
 };
+#[cfg(feature = "base_node")]
+use crate::{
+    chain_storage::calculate_validator_node_mr,
+    transactions::{transaction_components::transaction_output::batch_verify_range_proofs, CryptoFactories},
+    KernelMmr,
+    MutableOutputMmr,
+};
 use crate::{
     consensus::{
         emission::{Emission, EmissionSchedule},
         ConsensusConstants,
         NetworkConsensus,
     },
     proof_of_work::DifficultyAdjustmentError,
-    transactions::{tari_amount::MicroTari, transaction_components::TransactionKernel},
+    transactions::{
+        tari_amount::MicroTari,
+        transaction_components::{TransactionError, TransactionKernel},
+    },
 };
 
+#[cfg(feature = "base_node")]
+pub const NOT_BEFORE_PROOF_BYTES_SIZE: usize = u16::MAX as usize;
```

Comment:
Its perhaps not too bad, but this is not really required. There is no free reign on this data vec. 
For randomx blocks, this data needs to deserialize to a monero block
and if its a sha3x block, its required to be empty. 

And for the genesis block, because they can only be created in code we can see how large they are, and its not open to manipulation by hostile agents

Reply:
Hmmm... I had an interesting chat with ChatGPT about this ;)

How many bytes would a digitized grayscale headline of 1st January 2000 of The New York Times consist of if we need to display it on a 1280 x 768 pixel screen after reconstruction? Let us assume the  headline is 

_"as I sip my drink, thoughts of esmeralda consume my mind, like a refreshing nourishing draught_

_The New York Times , 2000/01/01_

_Lorem Ipsum_

_Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."_

Without thinking too much about it, the response seemed plausible, so as a thought experiment I chose `u16::MAX` as the prediction was `40,320` .

**Response**

To estimate the number of bytes a digitized grayscale headline of January 1, 2000, from The New York Times would occupy, considering the provided headline text and a screen resolution of 1280 x 768 pixels, we need to account for the following factors:

- Text Size and Font: The size and font used to display the headline will impact the number of pixels required for each character.

- Length of the Headline: The number of characters in the headline will contribute to its size since each character will require corresponding pixels in the grayscale representation.

- Resolution: The headline will be displayed on a 1280 x 768 pixel screen.

- Grayscale Encoding: Grayscale images typically use one byte (8 bits) per pixel to represent the intensity level of gray.

Considering the specific headline text and assuming an average font size, we can estimate the dimensions occupied by the headline on the screen.

Taking into account that the headline spans three lines, we can approximate the width to be around 1120 pixels (assuming an average of 8 pixels per character) and the height to be around 36 pixels (assuming 12 pixels per line). Therefore, the headline would require approximately 40,320 pixels (1120 pixels width * 36 pixels height).

Since grayscale images use one byte per pixel, we can estimate that the headline would consist of approximately 40,320 bytes (40,320 pixels * 1 byte/pixel).

Please note that this is a rough estimation, and the actual byte count may vary depending on the specific font, font size, text content, and compression techniques applied. For a more accurate estimation, it would be necessary to have specific details about the font, headline text, and rendering process, as well as consider any compression or encoding methods used.

Reply:
You don't want a digitised version. That's too hard to verify (download, convert save as a file, with right extension, view). 

KISS. Just place a suitable headline text in as a string.

Reply:
Well, in that case, we could limit the vector size all the way down to something like 2000, but I guess the specific number can be agreed upon later.